pkg/detect/util/detect.go

```diff
@@ -0,0 +1,151 @@
+package util
+
+import (
+	"context"
+	"maps"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+
+	dataTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data"
+	conditionTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/condition"
+	criteriaTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/condition/criteria"
+	criterionTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/condition/criteria/criterion"
+	ecosystemTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/data/detection/segment/ecosystem"
+	sourceTypes "github.com/MaineK00n/vuls-data-update/pkg/extract/types/source"
+	db "github.com/MaineK00n/vuls2/pkg/db/common"
+	detectTypes "github.com/MaineK00n/vuls2/pkg/detect/types"
+)
+
+type Request struct {
+	RootID  dataTypes.RootID
+	Query   criterionTypes.Query
+	Indexes []int
+}
+
+func Detect(dbc db.DB, ecosystem ecosystemTypes.Ecosystem, queries []string, createRequestFn func(rootID dataTypes.RootID, queries []string) Request, concurrency int) (map[dataTypes.RootID]detectTypes.VulnerabilityDataDetection, error) {
+	m, err := dbc.GetIndexes(ecosystem, queries...)
+	if err != nil {
+		return nil, errors.Wrap(err, "get indexes")
+	}
+
+	reqChan := make(chan Request, len(m))
```

Comment:
Is less capacity enough?? Because the producer is another goroutine, it pushes requests as workers pull them. Extremely, the capacity of 1 will be good.  Practically, the capacity of concurrency is better?

Reply:
changed: https://github.com/MaineK00n/vuls2/compare/b72b6f1dc37dc2a446696d6acdcfb9a8dccbd51c..81cc9fa5861836bedb38f28194e99051fd638d75